src/query/src/sql.rs

```diff
@@ -242,6 +280,176 @@ async fn query_from_information_schema_table(
+            .otherwise(lit(""))
+            .context(error::PlanSqlSnafu)?
+            .alias(COLUMN_KEY_COLUMN),
+        Expr::Wildcard,
+    ];
+
+    let projects = if stmt.full {
+        vec![
+            (columns::COLUMN_NAME, FIELD_COLUMN),
+            (columns::DATA_TYPE, COLUMN_TYPE_COLUMN),
+            (COLUMN_COLLATION_COLUMN, COLUMN_COLLATION_COLUMN),
+            (columns::IS_NULLABLE, COLUMN_NULLABLE_COLUMN),
+            (COLUMN_KEY_COLUMN, COLUMN_KEY_COLUMN),
+            (columns::COLUMN_DEFAULT, COLUMN_DEFAULT_COLUMN),
+            (columns::COLUMN_COMMENT, COLUMN_COMMENT_COLUMN),
+            (COLUMN_PRIVILEGES_COLUMN, COLUMN_PRIVILEGES_COLUMN),
+            (COLUMN_EXTRA_COLUMN, COLUMN_EXTRA_COLUMN),
+        ]
+    } else {
+        vec![
+            (columns::COLUMN_NAME, FIELD_COLUMN),
+            (columns::DATA_TYPE, COLUMN_TYPE_COLUMN),
+            (columns::IS_NULLABLE, COLUMN_NULLABLE_COLUMN),
+            (COLUMN_KEY_COLUMN, COLUMN_KEY_COLUMN),
+            (columns::COLUMN_DEFAULT, COLUMN_DEFAULT_COLUMN),
+            (COLUMN_EXTRA_COLUMN, COLUMN_EXTRA_COLUMN),
+        ]
+    };
+
+    let filters = vec![
+        col(columns::TABLE_NAME).eq(lit(&stmt.table)),
+        col(columns::TABLE_SCHEMA).eq(lit(schema_name.clone())),
+        col(columns::TABLE_CATALOG).eq(lit(query_ctx.current_catalog())),
+    ];
+    let like_field = Some(columns::COLUMN_NAME);
+    let sort = vec![col(columns::COLUMN_NAME).sort(true, true)];
+
+    query_from_information_schema_table(
+        query_engine,
+        catalog_manager,
+        query_ctx,
+        COLUMNS,
+        select,
+        projects,
+        filters,
+        like_field,
+        sort,
+        stmt.kind,
+    )
+    .await
+}
+
+/// Execute `SHOW INDEX` statement.
+pub async fn show_index(
+    stmt: ShowIndex,
+    query_engine: &QueryEngineRef,
+    catalog_manager: &CatalogManagerRef,
+    query_ctx: QueryContextRef,
+) -> Result<Output> {
+    let schema_name = if let Some(database) = stmt.database {
+        database
+    } else {
+        query_ctx.current_schema().to_owned()
+    };
+
+    let select = vec![
+        // 1 as `Non_unique`: contain duplicates
+        lit(1).alias(INDEX_NONT_UNIQUE_COLUMN),
+        // How the column is sorted in the index: A (ascending).
+        lit("A").alias(COLUMN_COLLATION_COLUMN),
+        null().alias(INDEX_CARDINALITY_COLUMN),
+        null().alias(INDEX_SUB_PART_COLUMN),
+        null().alias(INDEX_PACKED_COLUMN),
+        // case `constraint_name`
+        //    when 'TIME INDEX' then 'NO'
+        //    else 'YES'
+        // end as `Null`
+        case(col(key_column_usage::CONSTRAINT_NAME))
+            .when(lit(TIME_INDEX), lit(NO_STR))
+            .otherwise(lit(YES_STR))
+            .context(error::PlanSqlSnafu)?
+            .alias(COLUMN_NULLABLE_COLUMN),
+        // TODO(dennis): maybe 'BTREE'?
+        lit("greptime-inverted-index-v1").alias(INDEX_INDEX_TYPE_COLUMN),
```

Comment:
We have multiple indices enabled for a column. I'm not sure if it's a good idea to fix a literal value here.  I'm fine with current implement since no one is likely to depend on this result, otherwise we'd better to leave it empty.

Reply:
It's the index type, just an index information.  I think no one would depend on this column.